packages/kbn-unified-data-table/src/components/data_table_additional_display_settings.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { EuiFormRow, EuiFieldNumber } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { debounce } from 'lodash';
+
+export const MAX_ALLOWED_SAMPLE_SIZE = 1000;
+
+export interface UnifiedDataTableAdditionalDisplaySettingsProps {
+  sampleSize: number;
+  onChangeSampleSize: (sampleSize: number) => void;
+}
+
+export const UnifiedDataTableAdditionalDisplaySettings: React.FC<
+  UnifiedDataTableAdditionalDisplaySettingsProps
+> = ({ sampleSize, onChangeSampleSize }) => {
+  const [activeSampleSize, setActiveSampleSize] = useState<number>(sampleSize);
+
+  const debouncedOnChangeSampleSize = useMemo(
+    () => debounce(onChangeSampleSize, 300, { leading: false, trailing: true }),
+    [onChangeSampleSize]
+  );
+
+  const onChangeActiveSampleSize = useCallback(
+    (event) => {
+      const newSampleSize = Number(event.target.value);
+      if (newSampleSize > 0) {
+        setActiveSampleSize(newSampleSize);
+        if (newSampleSize <= MAX_ALLOWED_SAMPLE_SIZE) {
+          debouncedOnChangeSampleSize(newSampleSize);
+        }
+      }
+    },
+    [setActiveSampleSize, debouncedOnChangeSampleSize]
+  );
+
+  const sampleSizeLabel = i18n.translate('unifiedDataTable.sampleSizeSettings.sampleSizeLabel', {
+    defaultMessage: 'Sample size',
+  });
+
+  useEffect(() => {
+    setActiveSampleSize(sampleSize); // reset local state
+  }, [sampleSize, setActiveSampleSize]);
+
+  return (
+    <EuiFormRow label={sampleSizeLabel} display="columnCompressed">
+      <EuiFieldNumber
+        compressed
+        fullWidth
+        min={1}
+        max={MAX_ALLOWED_SAMPLE_SIZE}
```

Comment:
@opauloh Thanks for your feedback! The slider was not really helpful when having a step of 1, so an input would work better. But there is another option: to have a slider with a step 10 starting from 10. Updated the PR via https://github.com/elastic/kibana/pull/157269/commits/7feb2aeef7f9ba34465329c40a240def16352141

<img width="400" alt="Screenshot 2023-10-09 at 11 10 52" src="https://github.com/elastic/kibana/assets/1415710/e3b51581-b694-4ae3-9220-deeedd869998">

Reply:
LGTM 🚀

Thanks for updating it @jughosta ! I liked how it's visually clear now that it can go up to the max limit, and that the default selection is in between! 

Agree on changing the steps, a step of 10 makes way more sense for this purpose and improves the usability!